sdk/communication/azure-communication-administration/src/samples/java/com/azure/communication/administration/ReadmeSamples.java

```diff
@@ -108,4 +114,230 @@ public void deleteUser() {
+
+        // Create an HttpClient builder of your choice and customize it
+        HttpClient httpClient = new NettyAsyncHttpClientBuilder().build();
+
+        CommunicationClientCredential credential = new CommunicationClientCredential(accessToken);
+        HttpPipelinePolicy hmacAuthenticationPolicy = new HmacAuthenticationPolicy(credential);
+
+        PhoneNumberClient phoneNumberClient = new PhoneNumberClientBuilder()
+            .endpoint(endpoint)
+            .credential(credential)
+            .addPolicy(hmacAuthenticationPolicy)
+            .httpClient(httpClient)
+            .buildClient();
+
+        return phoneNumberClient;
+    }
+
+    /**
+     * Sample code to get all supported countries
+     *
+     * @return supported countries
+     */
+    public List<PhoneNumberCountry> getSupportedCountries() {
+        String locale = "en-us";
+
+        try {
+            PhoneNumberClient phoneNumberClient = createPhoneNumberClient();
+
+            List<PhoneNumberCountry> phoneNumberCountries = phoneNumberClient
+                .listAllSupportedCountries(locale)
+                .stream()
+                .collect(Collectors.toList());
+
+            for (PhoneNumberCountry phoneNumberCountry:
+                phoneNumberCountries) {
+                System.out.println("Phone Number Country Code: " + phoneNumberCountry.getCountryCode());
+                System.out.println("Phone Number Country Name: " + phoneNumberCountry.getLocalizedName());
+            }
+
+            return phoneNumberCountries;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    /**
+     * Sample code to get a list of all acquired phone numbers
+     *
+     * @return the acquired phone numbers
+     */
+    public List<AcquiredPhoneNumber> getAcquiredPhoneNumbers() {
+        String locale = "en-us";
+
+        try {
+            PhoneNumberClient phoneNumberClient = createPhoneNumberClient();
+
+            List<AcquiredPhoneNumber> acquiredPhoneNumbers = phoneNumberClient
+                .listAllPhoneNumbers(locale)
+                .stream()
+                .collect(Collectors.toList());
+
+            for (AcquiredPhoneNumber acquiredPhoneNumber:
+                 acquiredPhoneNumbers) {
+               System.out.println("Acquired Phone Number: " + acquiredPhoneNumber.getPhoneNumber());
+            }
+
+            return acquiredPhoneNumbers;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    /**
+     * Sample code to get a list of all phone plan groups
+     *
+     * @return phone plans groups
+     */
+    public List<PhonePlanGroup> getPhonePlanGroups() {
+        String countryCode = "US";
+        String locale = "en-us";
+
+        try {
+            PhoneNumberClient phoneNumberClient = createPhoneNumberClient();
+
+            List<PhonePlanGroup> phonePlanGroups = phoneNumberClient
```

Comment:
Since phoneNumberClient.listPhonePlanGroups returns an Iterable, you should be able to use the for loop on it directly.

```
PagedIterable<PhonePlanGroup> pagedIterable = phoneNumberClient.listPhonePlanGroups(countryCode, locale, true);
for (PhonePlanGroup phonePlanGroup: pagedIterable) {
    System.out.println("Phone Plan GroupId: " + phonePlanGroup.getPhonePlanGroupId());
    System.out.println("Phone Plan NumberType: " + phonePlanGroup.getPhoneNumberType());
}
```

We can do the same for all the other examples instead of collecting to a list first.

Reply:
Sure, that makes sense. Done